### ISSUE TITLE
Update to OCC 7.7.0

### ISSUE THREAD

**Issue by luzpaz** [FreeCAD/FreeCAD-snap#69]
Per [wwmayer](https://forum.freecadweb.org/viewtopic.php?p=639505#p639505) OCC 7.7.0 compiles fine with latest FreeCAD HEAD. We should add it to snap edge in that case

**Comment by adrianinsaval** [comment]
link to forum discussion: https://forum.freecadweb.org/viewtopic.php?p=639505#p639505

**Comment by adrianinsaval** [comment]
you'll need to change this to V7_7_1_BF
https://github.com/FreeCAD/freecad-deps-core20-snap/blob/d58e75c254cb308d46dc522a6a3211767b092725/snap/snapcraft.yaml#L25

it's possible that you will need to move to core 22 also:
https://github.com/FreeCAD/freecad-deps-core20-snap/blob/d58e75c254cb308d46dc522a6a3211767b092725/snap/snapcraft.yaml#L2

occt 7.7 requires tbb >=2021.5 which I doubt would be in ubuntu 20.04

**Comment by luzpaz** [comment]
> occt 7.7 requires tbb >=2021.5 which I doubt would be in ubuntu 20.04

You're right, @adrianinsaval. Do you know if I can keep things as they are for stable and change OCC and core 22  just for snap edge and snap beta builds?

<a href="https://repology.org/project/onetbb/versions">
  <img src="https://repology.org/badge/version-for-repo/ubuntu_20_04/onetbb.svg" alt="Ubuntu 20.04 package"></a>
<a href="https://repology.org/metapackage/onetbb/versions">
      <img src="https://repology.org/badge/vertical-allrepos/onetbb.svg?header=onetbb" alt="Packaging status" align="right"></a>

**Comment by adrianinsaval** [comment]
I would expect yes, but I'm not all that familiar with how  making snaps work

**Comment by adrianinsaval** [comment]
There's some bug with occt 7.7.1 that I don't think we're going to fix before release, I'll move this out of the 0.21 project. Better to do this next cycle.

**Comment by luzpaz** [comment]
7.7.2 released: https://dev.opencascade.org/content/open-cascade-technology-772-maintenance-release

**Comment by chennes** [comment]
Are we not building Blobfish for the snaps? I should fix if not.

**Comment by adrianinsaval** [comment]
we are using blobfish: https://github.com/FreeCAD/freecad-deps-snap/blob/d58e75c254cb308d46dc522a6a3211767b092725/snap/snapcraft.yaml#L24-L25

**Comment by luzpaz** [comment]
has blobfish rebased his patches on 7.7.2 yet ?

**Comment by chennes** [comment]
I don't think so: the latest I see is 7.7.1BF
https://gitlab.com/blobfish/occt/-/tree/V7_7_1_BF?ref_type=heads

**Comment by adrianinsaval** [comment]
the blobfish patch set keeps growing but none of it is going upstream, should we worry about that? Can we do something to get those merged upstream?

**Comment by luzpaz** [comment]
Maybe we can forge a relationship with upstream (there is a gentleman that has represented OCC in previous FLOSS conferences) and have a (paid?) liason work to merge patches ?

**Comment by chennes** [comment]
We cannot "merge" his code upstream, he has not signed the OpenCASCADE developer agreement, and is not interested in doing so. His patches would have to be rewritten.

**Comment by furgo16** [comment]
Should this issue be reworded to update the snap to 7.8.1?

https://gitlab.com/blobfish/occt/-/tree/V7_8_1?ref_type=tags

The Debian package on the daily PPA already seems to be using 7.8.1.

**Comment by luzpaz** [comment]
I renamed the ticket. Maybe I should of closed and opened a new one. Whatever.

I think @adrianinsaval was waiting on migrating to OCCT 7.8.1 until after the release. But I'm not remembering exactly the case.
